### ISSUE TITLE
CNAME data source fails if not resolvable to a host

### ISSUE THREAD

**Issue by BrandonALXEllisSS** [hashicorp/terraform-provider-dns#171]
### Affected Resource(s)
dns_cname_record_set data source

### Terraform Configuration Files
```hcl
data "dns_cname_record_set" "record" {
  host = "some.cname.entry.that.exists.but.doesnt.resolve"
}
```

### Output
```hcl
Error: error looking up CNAME records for "some.cname.entry.that.exists.but.doesnt.resolve": lookup some.cname.entry.that.exists.but.doesnt.resolve on 172.28.16.1:53: no such host
```

### Expected Behavior
Data source does not try to resolve CNAME entry and just returns the record

### Actual Behavior
Data source retrieves CNAME entry and then tries to resolve it, ultimately failing

### References
This is because the[ data source uses the go net provider's LookupCNAME function](https://github.com/hashicorp/terraform-provider-dns/blob/7ea5284917be8485ca9f58a5184e85ce7b4024d9/internal/provider/data_dns_cname_record_set.go#L32). And for some reason, [this provider has this functionality](https://stackoverflow.com/questions/56856075/how-can-i-get-the-cname-of-a-host-for-which-dns-resolution-fails-nxdomain-in-g). Consider changing this so that it makes sense please

**Comment by drdamour** [comment]
yeah the cname lookup should leverage https://godoc.org/github.com/miekg/dns  the way it is now it doesn't even respect the dns server config in the provider, it just uses the host machines dns
